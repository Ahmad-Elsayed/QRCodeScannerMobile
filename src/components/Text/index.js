import React from 'react';
import { Text as RNText } from 'react-native';
import styles from './styles';

const Text = ({ style, children }) => {
  return <RNText style={[styles.textStyle, style]}>{children}</RNText>;
};

export default Text;
