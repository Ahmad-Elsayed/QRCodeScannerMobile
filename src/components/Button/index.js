import React from 'react';
import Text from '../Text';
import styles from './styles';
import { Pressable } from 'react-native';

const Button = ({ title, onPress, containerStyle }) => {
  return (
    <Pressable
      onPress={onPress}
      style={({ pressed }) => [
        styles.containerStyle,
        pressed && styles.pressedStyle,
        containerStyle,
      ]}>
      <Text style={styles.textStyle}>{title}</Text>
    </Pressable>
  );
};

export default Button;
