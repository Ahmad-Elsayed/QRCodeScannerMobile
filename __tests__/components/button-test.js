/**
 * @format
 */

import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import { Button } from 'components';

it('renders correctly', () => {
  const button = renderer.create(<Button title="test" />).toJSON();
  expect(button).toMatchSnapshot();
});
