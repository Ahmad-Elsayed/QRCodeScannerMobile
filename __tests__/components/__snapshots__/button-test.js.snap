// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#45b1e8",
        "borderRadius": 6,
        "paddingHorizontal": 30,
        "paddingVertical": 12,
      },
      false,
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
          "textAlign": "left",
        },
        Object {
          "color": "#fff",
          "fontSize": 13,
          "fontWeight": "500",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
